src/App.js

```diff
@@ -1,15 +1,107 @@
 import React, { Component } from 'react';
+import ContactList from './components/contact-list.js';
+import SearchBar from './components/search-bar.js';
+import Button from './components/button.js';
+import axios from 'axios';
+
+const data = {
+  contacts: [
+    {
+      _id: 1,
+      name: '',
+      occupation: '',
+      avatar: ''
+    },
+  ]
+};
+
+export default class App extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      data,
+      query: '',
+      selected: ['No Contacts Selected']
+    };
+  }
+
+  componentDidMount() {
+    axios.get('http://localhost:4000/contacts')
+      .then(resp => {
+        data.contacts = resp.data;
+        this.setState({
+          data
+        });
+      })
+      .catch(err => {
+        console.log(`Error! ${err}`);
+      });
+  }
+
 
-class App extends Component {
   render() {
     return (
       <div className="App">
         <h1>
-          Contact List!
+          <SearchBar value={this.state.query} handleSearch={this.handleSearch.bind(this)} />
+          <ContactList data={this.getFilteredContacts()} handleSelect={this.handleSelect.bind(this)} searchValue={this.state.query} />
         </h1>
+        <h1>Selected Contacts </h1>
+        <ContactList data={this.getSelectedContacts()} handleSelect={this.handleSelect.bind(this)} searchValue={this.state.query} />
+        <Button resetSelection={this.resetSelection.bind(this)} />
       </div>
     );
   }
-}
 
-export default App;
+  resetSelection() {
+
+    this.setState(
+        this.state.data.contacts.map((obj) => {
+          obj.isSelected = false;
+        })
```

Comment:
setState always takes an **Object** as an argument. You are passing an array (the return value of the map function). This only works, because you are mutating the state directly through `obj.isSelected`. Mutation of the state should not be used at all in react. If you want to change something, you have to clone the objects. 
You are making this harder than it is because you are storing everything in a data object inside of state. Better keep the state object flat and store everything on it directly.